### PR TITLE
[JENKINS-68293] Login and signup page refresh

### PR DESCRIPTION
**Pull request by jamesD33064**
See [JENKINS-68293](https://issues.jenkins-ci.org/browse/JENKINS-68293).

<details>
<summary>Login page before</summary>

image

</details>

<details>
<summary>Login page after</summary>

image

</details>

<details>
<summary>Signup page before</summary>

image

</details>

<details>
<summary>Signup page after</summary>

image

</details>

### Proposed changelog entries

* Login and signup page refresh

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
Being discussed in https://issues.jenkins.io/browse/JENKINS-66630

**Review by oleg-nenashev — COMMENTED** [review]
No strong opinion. I'd guess one of the UX SIG meetings could make a call on this

**Comment by alecharp** [comment]
@janfaracik @uhafner @timja I don't know if that was discussed during a UX SIG meeting, but I feel like it's a good thing to have the login button after the "keep me signed in" checkbox. The discussion on the ticket stopped last september. So, either we go forward with this, or we close this PR. 

I'm for accepting it, based on the justification on the ticket and what seems to be in place in other services.

**Review by alecharp — APPROVED** [review]

**Comment by timja** [comment]
it was never discussed at a UX meeting.

Given this doesn't seem to be a widely used pattern anymore a more complete fix would evaluate either removing the functionality or checking it by default but 🤷 . This is probably okay

Although I think the checkbox should be aligned with the other elements. It was wrong before but it's worse now.

**Review by NotMyFault — APPROVED** [review]
+0 from my side.

A "remember me" button you have to tick frequently by hand feels kinda old fashioned to me. In the future, the current implementation could be superseded by an Azure-like approach, if we want to keep the button, which asks you one time whether you want to stay logged in allowing you to change it later in your settings or something completely different, that gets a rid of the button at all while keeping the functionality in the background, e.g. like much other (web) services nowadays.

**Comment by alecharp** [comment]
@jamesD33064 could you look into the comment from Tim about the alignment of the field? Thanks.

**Comment by basil** [comment]
I could accept the argument that the "Keep me signed in" should be before the "Sign in" button, but the implementation provided in this PR with the "Keep me signed in" section centered looks odd to me. I can't give a logical argument here, but it just feels off balance. Compare with these links:

- https://phppot.com/wp-content/uploads/2016/03/php-login-script-with-remember-me.png
- https://www.codejava.net/images/articles/frameworks/springboot/remember-me/Default_login_page_with_Remember_me_option.png
- https://cloudbees.okta.com

In all cases the "Keep me signed in" checkbox is indeed before the "Sign in" button, but also it is left aligned rather than centered. To me, this feels more balanced from a subjective point of view.

I would be fine with approving this PR if someone could push a commit left-aligning the "Keep me signed in" section.

**Comment by timja** [comment]
I’m working on this currently, should have something today / tomorrow.

(Alignment is easy but I’m also changing it to use the same styles as the rest of core, the current checkbox these pages use has no focus state which moving it higher up really highlights)

**Comment by timja** [comment]
the login and signup page now uses the same CSS modules as the rest of core which fixed a few issues.

**Review by basil — COMMENTED** [review]
Very nice! I think the login page looks much better with the "Keep me signed in" box left-aligned.

The old login screen was 300px wide, but the new one is 236px wide. I cannot see any reason for shrinking it. The new one feels cramped to me.

Also the "Your browser will reload automatically when Jenkins is ready" text was nicely centered in the old version but seems oddly left-aligned in the new version.

(Something feels off to me about the vertical padding above and below "Please sign in below or create an account" / "If you already have a Jenkins account please sign in" in the new version, but I cannot quite articulate it. I will think about this some more.)

**Comment by janfaracik** [comment]
My local instance of Jenkins is a little corrupt at the moment - but  I'll give this a spin as soon as possible, looks great :)

**Comment by timja** [comment]
> My local instance of Jenkins is a little corrupt at the moment - but I'll give this a spin as soon as possible, looks great :)

core pr tester: :)
```command
docker run --rm -ti -p 8080:8080 -e ID=5743 -e MERGE_WITH=master jenkins/core-pr-tester
```

**Review by NotMyFault — APPROVED** [review]
Looks much better now, great work 🎉

File: core/src/main/resources/jenkins/model/Jenkins/login.jelly
Comment (janfaracik) [inline core/src/main/resources/jenkins/model/Jenkins/login.jelly:75]: Would be nice if `.signuptag` had a `margin-top: 0` and a `margin-bottom: 2rem` - balances it out a little better 
image

Comment (janfaracik) [reply core/src/main/resources/jenkins/model/Jenkins/login.jelly:75]: Same for the "If you have a Jenkins account..." and "A strong password is a long password th...." 

image

**Comment by janfaracik** [comment]
This looks awesome - great job :)

**Review by janfaracik — APPROVED** [review]

**Review by basil — APPROVED** [review]
Looks great! Very nice!

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by timja** [comment]
> Is this a synonym?

Yes it's a synonym. Not sure why it differs, at a guess it happened when the login pages were redone in the 'simple page' layout

**Comment by basil** [comment]
Jakob Nielsen and Marie Tahir's book [_Homepage Usability_](https://www.nngroup.com/books/homepage-usability/) (New Riders, 2002) recommends the use of "sign in" / "sign out" over "log in" / "log out" based on a survey of popular websites of that time. I have no strong preference between one usage or the other, but it would be nice to be consistent. Anyway, that is a pre-existing issue so I don't feel strongly that it be addressed in this PR.

**Pull request merged**
